airflow/utils/state.py

```diff
@@ -192,7 +193,7 @@ def color_fg(cls, state):
     """
 
     success_states: FrozenSet[TaskInstanceState] = frozenset(
-        [TaskInstanceState.SUCCESS, TaskInstanceState.SKIPPED]
+        [TaskInstanceState.SUCCESS, TaskInstanceState.SKIPPED, TaskInstanceState.REMOVED]
```

Comment:
This one I'm not so sure of. Adding it to finished states seems fine, but this one feels like it's asking for trouble.

Reply:
Is this one needed @ephraimbuddy? From a _very_ quick look at the code I would have thought that adding REMOVED to finished states would be enough for the DagRun to make it to success

Reply:
Oh I mis-read it. If you don't put this here then the DagRun is detected as deadlocked isn't it?

Reply:
It's not detected as deadlock but it keeps on trying to restore it and the dagrun would be stuck in running state. In dagrun.update_states, even if the task instance state is in finished states, they still need to be in either failed states or success states to determine the overall state of the dagrun

Reply:
Applied the suggestion and added a test